fili-core/src/main/java/com/yahoo/bard/webservice/web/handlers/workflow/DruidWorkflow.java

```diff
@@ -115,14 +117,15 @@ public DataRequestHandler buildWorkflow() {
         }
 
         // If query caching is enabled, the cache is checked before sending the request
-        if (BardFeatureFlag.DRUID_CACHE.isOn()) {
-            if (BardFeatureFlag.DRUID_CACHE_V2.isOn()) {
-                uiHandler = new CacheV2RequestHandler(uiHandler, dataCache, querySigningService, mapper);
-                nonUiHandler = new CacheV2RequestHandler(nonUiHandler, dataCache, querySigningService, mapper);
-            } else {
-                uiHandler = new CacheRequestHandler(uiHandler, dataCache, mapper);
-                nonUiHandler = new CacheRequestHandler(nonUiHandler, dataCache, mapper);
-            }
+        if (CacheFeatureFlag.TTL.isOn()) {
+            uiHandler = new CacheRequestHandler(uiHandler, dataCache, mapper);
+            nonUiHandler = new CacheRequestHandler(nonUiHandler, dataCache, mapper);
+        } else if (CacheFeatureFlag.LOCAL_SIGNATURE.isOn()) {
+            uiHandler = new CacheV2RequestHandler(uiHandler, dataCache, querySigningService, mapper);
+            nonUiHandler = new CacheV2RequestHandler(nonUiHandler, dataCache, querySigningService, mapper);
+        } else if (CacheFeatureFlag.ETAG.isOn()) {
+            uiHandler = new EtagCacheRequestHandler(uiHandler, dataCache, querySigningService, mapper);
+            nonUiHandler = new EtagCacheRequestHandler(uiHandler, dataCache, querySigningService, mapper);
```

Comment:
We should implement in this PR since the code will break, and this PR should begin allowing people to use the cache. @michael-mclawhorn  what do you think?